src/Apps/Sell/Routes/ArtistNotEligibleRoute.tsx

```diff
@@ -20,6 +24,32 @@ interface ArtistNotEligibleRouteProps {
 
 export const ArtistNotEligibleRoute: React.FC<ArtistNotEligibleRouteProps> = props => {
   const artist = useFragment(FRAGMENT, props.artist)
+  const { showAuthDialog } = useAuthDialog()
+  const { router } = useRouter()
+  const { isLoggedIn } = useSystemContext()
+
+  const onAddToMyCollection = () => {
+    if (isLoggedIn) {
+      router.push("/collector-profile/my-collection/artworks/new")
+    } else {
```

Comment:
Yeah, I feel like early returns are useful when there's much more of the function definition to skip, but in this case it's clear that there are only two paths for the function flow.

Reply:
I completely agree, it's pretty clear how it is. If there are just two paths, there is no need to use an early return.

Another argument, though, for early returns (even in this case) is to "highlight" the happy path, which is the only one that is not an early return. In this case, it would be `router.push("/collector-profile/my-collection/artworks/new")`.

Reply:
Switched to an early exit but also extracted the showAuthDialog behavior to keep things readable. 👍🏼